src/Service/BroadcastsService.php

```diff
@@ -84,4 +87,41 @@ function () use ($serviceId, $startDate, $endDate, $limit, $page) {
             }
         );
     }
+
+    /**
+     * @return Broadcast[]
+     */
+    public function findUpcomingByService(
+        Service $service,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL
+    ): array {
+        $key = $this->cache->keyHelper(
+            __CLASS__,
+            __FUNCTION__,
+            $service->getDbId(),
+            $limit,
+            $page,
+            $ttl
+        );
+
+        // TODO set the cache time based on the end date of the first returned item, so that the cache expires once the
+        // as soon as the current programme changes to something else
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () use ($service, $limit, $page) {
+                $dbEntities = $this->repository->findUpcomingByService(
+                    $service->getDbId(),
+                    'Broadcast',
+                    ApplicationTime::getTime(),
```

Comment:
How does this play with timezones? Should it be `ApplicationTime::getLocalTime()`

Reply:
I _think_ this should be OK as it is. It will probably need some logic in the frontend side.